lsl/imaging/utils.py

```diff
@@ -1541,6 +1514,80 @@ def bm_image(self, center=(0,0), term=None, weighting='natural', local_fraction=
+            
+    @property
+    def mjd(self):
+        """
+        Observation/image MJD.
+        """
+        
+        return self._mjd
+        
+    @mjd.setter
+    def mjd(self, value):
+        if isinstance(value, AstroTime):
+            self._mjd = value.utc.mjd
+        elif isinstance(value, (int, np.integer, float, np.floating)):
+            self._mjd = float(value)
+        else:
+            raise TypeError("Expected an object of type astropy.time.Time, int, or float")
+            
+    @property
+    def phase_center(self):
+        """
+        Observation/image phase center.
+        """
+        
+        return self._phase_center
+        
+    @phase_center.setter
+    def phase_center(self, value):
+        if isinstance(value, (SkyCoord, ICRS, FK5, ephem.Body, astro.equ_posn)):
+            self._phase_center = value
+        else:
+            raise TypeError("Expected an object of type SkyCoord, ICRS, FK5, ephem.Body, or astro.equ_posn")
+            
+    @property
+    def wcs(self):
+        """
+        World Coordinate System (WCS) associated with the image in equatorial
+        (RA/dec.) coordinates.
+        """
+        
+        if self._mjd is None:
+            raise RuntimeError("Cannot create WCS: unknown observation MJD")
+        if self._phase_center is None:
+            raise RuntimeError("Cannot create WCS: unknown phase center")
+            
+        # Extract what we need from the phase center
+        obs_t = AstroTime(self._mjd, format='mjd', scale='utc')
+        if isinstance(self._phase_center, astro.equ_posn):
+            pc_ra = self._phase_center.ra
+            pc_dec = self._phase_center.dec
+        else:
+            try:
+                pc = self._phase_center.transform_to(FK5(equinox=obs_t))
+                pc_ra = pc.ra.deg
+                pc_dec = pc.dec.deg
+            except AttributeError:
+                pc_ra = np.degrees(self._phase_center.ra)
+                pc_dec = np.degrees(self._phase_center.dec)
+                
+        return WCS(header={'WCSAXES': 2,
+                           'RADESYS': 'FK5',
+                           'EQUINOX': obs_t.jyear,
+                           'CTYPE1':  'RA---SIN',
+                           'CRPIX1':  np.round(self.size/self.res) // 2 + 1,
+                           'CDELT1':  -1 * np.degrees(self.pixel_size),
+                           'CRVAL1':  pc_ra,
+                           'CUNIT1':  'deg',
+                           'CTYPE2':  'DEC--SIN',
+                           'CRPIX2':  np.round(self.size/self.res) // 2 + 1,
+                           'CDELT2':  np.degrees(self.pixel_size),
+                           'CRVAL2':  pc_dec,
+                           'CUNIT2':  'deg',
+                           'LONPOLE': 180.0,
+                           'LATPOLE': 90.0})
```

Comment:
Here's another place where we need to make sure this is correct.  However, with what's going on at https://github.com/lwa-project/orville_wideband_imager/issues/24 that may be easier said than done.